core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1415,10 +1415,11 @@ package object config {
 
   private[spark] val SHUFFLE_HOST_LOCAL_DISK_READING_ENABLED =
     ConfigBuilder("spark.shuffle.readHostLocalDisk")
-      .doc(s"If enabled (and `${SHUFFLE_USE_OLD_FETCH_PROTOCOL.key}` is disabled and external " +
-        s"shuffle `${SHUFFLE_SERVICE_ENABLED.key}` is enabled), shuffle " +
-        "blocks requested from those block managers which are running on the same host are read " +
-        "from the disk directly instead of being fetched as remote blocks over the network.")
+      .doc(s"If enabled (and `${SHUFFLE_USE_OLD_FETCH_PROTOCOL.key}` is disabled, shuffle " +
+        "blocks requested from those block managers which are running on the same host are " +
+        "read from the disk directly instead of being fetched as remote blocks over the " +
+        "network. Note that for k8s workloads, this only works when nodes are using " +
+        "non-isolated container storage.")
```

Comment:
This is correct, but shall we remove this comment which is specific to some resource manager? MESOS has been using Container for a long time, too.
```
Note that for k8s workloads, this only works when nodes are using non-isolated container storage.
```

Reply:
Good point 👍

Reply:
@Ngone51 on containerized resource manager having a non-isolated container storage won't be enough, as for this feature to work we need to detect this non-isolation. Currently this is done by using the some host in the blockmanager ID which works only for YARN and standalone mode, is not it?
A question for the future: do you have a plan to introduce block manager grouping based on shared storage?

Reply:
> Currently this is done by using the some host in the blockmanager ID which works only for YARN and standalone mode, is not it?

IIUC, from @holdenk 's previous comment and @dongjoon-hyun 's comment, it should also work for Mesos/K8s when they're using the non-isolated container.


> A question for the future: do you have a plan to introduce block manager grouping based on shared storage?

I don't. To be honest, I'm not familiar with the containerized resource manager. I'm also not sure what the plan you're meaning here. Is it only needed for the containerized resource manager?

Reply:
Sorry for the confusion. 

My statement: this feature does not work on containerized systems even on non-isolated storage based containers as on those system AFAIK all the containers (running the executors and block managers along) have different hostnames (and IP addresses).

This feature currently is built on detecting the shared storage by the reused hostnames:
https://github.com/apache/spark/blob/6b97be552b1d5a78f476c4a97d794c567222d9bf/core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala#L307

Reply:
Thank you four your clarification. If that's the truth, I think we should point it out at doc. cc @dongjoon-hyun @holdenk